tabular/src/autogluon/tabular/trainer/abstract_trainer.py

```diff
@@ -1015,6 +1023,27 @@ def _train_and_save(self, X, y, model: AbstractModel, X_val=None, y_val=None, st
                 logger.exception('Detailed Traceback:')
             del model
         else:
+            if pruned_features:
+                original_model = self.load_model('_'.join(model.name.split('_')[:-1]))
+                leaderboard = self.leaderboard()
+                original_score = leaderboard[leaderboard['model'] == original_model.name]['score_val'].item()
+                score_str = f"({round(score, 4)} vs {round(original_score, 4)})"
+                if score > original_score:
+                    logger.log(30, f"Pruned model's score is better than original model's score {score_str}. Replacing original model...")
+                    self.delete_models(models_to_delete=original_model.name, dry_run=False)
+                    self._add_model(model=model, stack_name=stack_name, level=level)
+                    model_names_trained.append(model.name)
+                    self._debug_info['proxy_model'][-1]['score_improvement_from_proxy_yes'] += 1
+                else:
+                    logger.log(30, f"Pruned model's score is not better than original model's score {score_str}. Keeping original model...")
```

Comment:
I don't think "Pruned model" is clear enough terminology. This would be better replaced everywhere with something like "Model trained with feature pruning"